hvac/api/auth_methods/cert.py

```diff
@@ -0,0 +1,254 @@
+        Supported methods:
+            DELETE: /auth/{mount_point}/groups. Produces: 204 (empty body)
+
+        :param name: The name of the certificate role.
+        :type name: str | unicode
+        :param mount_point:
+        :type mount_point:
+        """
+        api_path = '/v1/auth/{mount_point}/certs/{name}'.format(mount_point=mount_point, name=name)
+        return self._adapter.delete(
+            url=api_path,
+        )
+
+    def configure_tls_certificate(self, mount_point='cert', disable_binding=False):
+        """
+        Configure options for the method.
+
+        Supported methods:
+            POST: /auth/<mount point>/config. Produces: 204 (empty body)
+
+
+        :param disable_binding: If set, during renewal, skips the matching of presented client identity with the client
+            identity used during login.
+        :type disable_binding: bool
+        :param mount_point:
+        :type mount_point:
+        """
+        params = {
+            'disable_binding': disable_binding,
+        }
+        api_path = '/v1/auth/{mount_point}/config'.format(mount_point=mount_point)
+        return self._adapter.post(
+            url=api_path,
+            json=params,
+        )
+
+    def login(self, name="", cacert=False, cert_pem="", key_pem="", mount_point='cert', use_token=True):
+        """
+        Log in and fetch a token. If there is a valid chain to a CA configured in the method and all role constraints
+            are matched, a token will be issued. If the certificate has DNS SANs in it, each of those will be verified.
+            If Common Name is required to be verified, then it should be a fully qualified DNS domain name and must be
+            duplicated as a DNS SAN
+
+        Supported methods:
+            POST: /auth/<mount point>/login Produces: 200 application/json
+
+        :param name: Authenticate against only the named certificate role, returning its policy list if successful. If
+            not set, defaults to trying all certificate roles and returning any one that matches.
+        :type name: str | unicode
+        :param cacert: The value used here is for the Vault TLS Listener CA certificate, not the CA that issued the
+            client authentication certificate. This can be omitted if the CA used to issue the Vault server certificate
+            is trusted by the local system executing this command.
+        :type cacert: str | bool
+        :param cert_pem: Location of the cert.pem used to authenticate the host.
+        :tupe cert_pem: str | unicode
+        :param key_pem: Location of the public key.pem used to authenticate the host.
+        :param key_pem: str | unicode
+        :param mount_point:
+        :type mount_point:
+        :param use_token: If the returned token is stored in the client
+        :param use_token: bool
+        :return: The response of the login request.
+        :rtype: requests.Response
+        """
+        params = {'use_token': use_token}
```

Comment:
Fixed in https://github.com/hvac/hvac/pull/720

Reply:
Good catch. The use_token was one of the things I hadn't originally had in scope as I wasn't originally planning on depricating the auth_tls method. I was originally only trying to create a method to create the certificate endpoints as well as allow an already created client to auth without the need to recreate it. This is because the certs are not saved as part of the adapter class and thus could only be used during instantiation of a new client object.